### ISSUE TITLE
(Suggestion) Include more useful parameters as form inputs in colab

### ISSUE THREAD

**Issue by afiaka87** [lucidrains/deep-daze#8]
I've included these form inputs in my personal copy of your Colab notebook. In particular, reducing the image_width parameter allows for one to vastly increase the number of hidden_layers. By going to an image_width of 256 (instead of the default 512) I was able to run 32 hidden layers without problems on a T4. 

```python
from tqdm import trange
from IPython.display import Image, display

from deep_daze import Imagine

TEXT = 'blue marshmallow' #@param {type:"string"}
NUM_LAYERS = 16 #@param {type:"number"}
SAVE_EVERY =  20#@param {type:"number"}
IMAGE_WIDTH = 512 #@param {type:"number"}
SAVE_PROGRESS = False #@param {type:"boolean"}
LEARNING_RATE = 1e-5 #@param {type:"number"}
ITERATIONS = 1050 #@param {type:"number"}

model = Imagine(
    text = TEXT,
    num_layers = NUM_LAYERS,
    save_every = SAVE_EVERY,
    image_width = IMAGE_WIDTH,
    lr = LEARNING_RATE,
    iterations = ITERATIONS,
    save_progress = SAVE_PROGRESS
)
```

Feel free to include them in your copy if you'd like to.

**Comment by afiaka87** [comment]
I will say that going too low with the image_width parameter (or too high with the learning rate parameter) produces mostly noise.

**Comment by lucidrains** [comment]
@afiaka87 ok added!

**Comment by afiaka87** [comment]
Hm, I'm curious actually. Did you intend to leave the default parameter for layers at 32? I assume this will fail for many with the IMG_WIDTH also set to 512. I've found 20 to be the upper limit at that resolution on the 16 GiB Tesla T4 colab gives me but 16 is probably the safer choice.

**Comment by lucidrains** [comment]
@afiaka87 which version of deep daze are you running? in the latest version, i do gradient accumulation, and it fits on my 1 11GB 2080Ti at home

**Comment by lucidrains** [comment]
@afiaka87 I lowered it to 20 - i'm using colab pro, so perhaps i have more resources than normal

**Comment by afiaka87** [comment]
> @afiaka87 which version of deep daze are you running? in the latest version, i do gradient accumulation, and it fits on my 1 11GB 2080Ti at home, even with 32 layers

Wow! You're right, I hadn't checked my memory usage recently enough and sure enough it's using substantially less now. Sorry, I should have researched the issue better. 32 is a fine default.

I'll close the issue, but I am curious about the gradient accumulation. What are the tradeoffs there?

**Comment by lucidrains** [comment]
@afiaka87 no tradeoff :) it's a nice optimization trick

**Comment by afiaka87** [comment]
I'm on colab pro myself.  As for non-pro users, I have a friend who worked on this notebook all day. He ran it all day and kept getting a T4 (16 GiB vram) which is obviously more than enough.  Your success on your 2080TI (nice card, btw) has inspired me to get it running locally on my 8GB 2070 Super.

**Comment by lucidrains** [comment]
@afiaka87 i barely squeak by with 10 / 11 GB usage, so you may have to tweak your layers, batch size, or gradient accumulation steps, but which are all accessible through the CLI https://github.com/lucidrains/deep-daze/blob/main/deep_daze/cli.py#L9-L11

**Comment by afiaka87** [comment]
Lowering the batch size won't effect the output will it? Just the length of time it takes?
